torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -775,13 +775,11 @@ def get_state_dict_type(module: nn.Module) -> StateDictSettings:
                 )
 
             # If device_mesh is passed in when initalizing FSDP, we automatically turn the
-            # _use_dtensor flag to be true for ShardedStateDictConfig() and ShardedOptimStateDictConfig().
-            if getattr(module, "device_mesh", None) and isinstance(
-                state_dict_settings.state_dict_config, ShardedStateDictConfig
-            ):
-                state_dict_settings.state_dict_config._use_dtensor = True
-            if getattr(module, "device_mesh", None) and isinstance(
-                state_dict_settings.optim_state_dict_config, ShardedOptimStateDictConfig
+            # _use_dtensor flag to be true for ShardedOptimStateDictConfig().
+            if (
```

Comment:
I know this was here before this diff, but since this is a get method, ideally it should avoid modifying state. So IMO we should see if we can avoid `use_dtensor = True` setting here entirely.

Reply:
> I know this was here before this diff, but since this is a get method, ideally it should avoid modifying state. So IMO we should see if we can avoid `use_dtensor = True` setting here entirely.

Good point. Thanks for bringing it up. Will add a todo to figure out a more elegant way to handle this.